features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/mapper/ImageViewMapper.kt

```diff
@@ -33,6 +36,29 @@ internal class ImageViewMapper(
     viewBoundsResolver,
     drawableToColorMapper
 ) {
+    private var imageViewUtils: ViewUtils = NoOpViewUtils()
+
+    init {
+        if (imageViewUtils is NoOpViewUtils) {
```

Comment:
are you doing this because you do not want to expose the `ImageViewUtils` ?

Reply:
Yes, exactly. While we have to make the mapper public there's no reason to also expose the utils (at least at the moment).

Reply:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ In that case why not making it:
```
private var imageViewUtils: ViewUtils = ImageViewUtils
    

    internal constructor(
        viewIdentifierResolver: ViewIdentifierResolver,
        colorStringFormatter: ColorStringFormatter,
        viewBoundsResolver: ViewBoundsResolver,
        drawableToColorMapper: DrawableToColorMapper,
        imageViewUtils: ImageViewUtils
    ) : this(
        viewIdentifierResolver,
        colorStringFormatter,
        viewBoundsResolver,
        drawableToColorMapper
    ) {
        this.imageViewUtils = imageViewUtils
    }
    
```

Reply:
This is possible, but it means that `ImageViewUtils` will always be initialized - even when that initialization would be unnecessary because something else was injected.

Reply:
I am not sure I get it here but let's discuss it a bit more tomorrow. The way I see it there's one constructor for us internally where we can pass our own instance of `ImageViewUtils` for testing purposes mainly and one for external usage where the `imageViewUtils` is initialized by default as `ImageViewUtils` singleton. Am I missing something ?